### PR TITLE
chore: refactoring image tagging reader service

### PR DESCRIPTION
**Pull request by Ash-exp**
# Description

This PR adds reader service for `ImageTaggingRepository`.

Fixes https://github.com/devtron-labs/sprint-tasks/issues/1110

## Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code.
* [x] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/devtronResource/history/deployment/cdPipeline/DeploymentHistoryService.go [hunk 1]
```diff
@@ -17,19 +18,22 @@ type DeploymentHistoryService interface {
 type DeploymentHistoryServiceImpl struct {
 	logger                              *zap.SugaredLogger
 	cdHandler                           pipeline.CdHandler
+	imageTaggingReadService             read.ImageTaggingReadService
 	imageTaggingService                 pipeline.ImageTaggingService
```
Comment (kartik-579) [inline pkg/devtronResource/history/deployment/cdPipeline/DeploymentHistoryService.go:22]: remove old import

File: pkg/pipeline/ImageTaggingService.go [hunk 2]
```diff
@@ -50,27 +51,26 @@ type ImageTaggingService interface {
...
 	GetTagsDataMapByAppId(appId int) (map[int][]*repository.ImageTag, error)
 	// GetImageCommentsDataMapByArtifactIds this will fetch a map of artifact vs imageComment for given artifactIds
 	GetImageCommentsDataMapByArtifactIds(artifactIds []int) (map[int]*repository.ImageComment, error)
-	// GetUniqueTagsByAppId gets all the unique tag names for the given appId
-	GetUniqueTagsByAppId(appId int) ([]string, error)
 	GetImageTaggingServiceConfig() ImageTaggingServiceConfig
 	FindProdEnvExists(externalCi bool, pipelineIds []int) (bool, error)
 }
 
 type ImageTaggingServiceImpl struct {
```
Comment (kartik-579) [inline pkg/pipeline/ImageTaggingService.go:62]: can move this service to new path , i.e. /build/artifacts/imageTagging

**Review by kartik-579 — CHANGES_REQUESTED** [review]

File: pkg/devtronResource/history/deployment/cdPipeline/DeploymentHistoryService.go (hunk 1 above)
Comment (Ash-exp) [reply pkg/devtronResource/history/deployment/cdPipeline/DeploymentHistoryService.go:22]: There is this method, using `imageTaggingService`. Ideally `GetProdEnvByCdPipelineId` should be the part of `CdPipelineReadService`
```go
prodEnvExists, err := impl.imageTaggingService.GetProdEnvByCdPipelineId(req.PipelineId)
```

File: pkg/pipeline/ImageTaggingService.go (hunk 2 above)
Comment (Ash-exp) [reply pkg/pipeline/ImageTaggingService.go:57]: updated 👍

**Review by nishant-d — DISMISSED** [review]

**Review by kartik-579 — APPROVED** [review]

**Review by vikramdevtron — APPROVED** [review]

**Pull request merged**